### ISSUE TITLE
podman sock not working in Fedora 41 image (rootful mode)

### ISSUE THREAD

**Issue by hasan4791** [lima-vm/lima#3038]
### Description

Podman sock isn't working in podman-rootful template having Fedora 41 image in my Mac M1 pro.

**Steps to reproduce:**
1. limactl create template://podman-rootful --vm-type vz --cpus 4 --memory 8 --network vzNAT --rosetta --mount-writable --mount-type virtiofs
2. podman system connection add lima-podman-rootful "unix:///Users/chasan/.lima/podman-rootful/sock/podman.sock"
3. podman system connection default lima-podman-rootful
```
 (chasan-working) vpn-proxy> podman images
Cannot connect to Podman. Please verify your connection to the Linux system using `podman system connection list`, or try `podman machine init` and `podman machine start` to manage a new Linux VM
Error: unable to connect to Podman socket: Get "http://d/v5.3.1/libpod/_ping": EOF
(chasan-working) vpn-proxy> 
```

**Workaround:**
Updated the following image in template and everything is working fine.
```
- location: "https://download.fedoraproject.org/pub/fedora/linux/releases/40/Cloud/aarch64/images/Fedora-Cloud-Base-Generic.aarch64-40-1.14.qcow2"
  arch: "aarch64"
  digest: "sha256:ebdce26d861a9d15072affe1919ed753ec7015bd97b3a7d0d0df6a10834f7459"
```

**Comment by afbjorklund** [comment]
You should be able to find more details in the log, it is not a generic problem with Podman 5.3.1 on Fedora 41

**Comment by AkihiroSuda** [comment]
Fedora 41 (Lima v1.0.2):
```console
$ limactl shell podman-rootful curl --unix-socket /run/podman/podman.sock http://foo/_ping
curl: (7) Failed to connect to foo port 80 after 0 ms: Could not connect to server

$ limactl shell podman-rootful sudo curl --unix-socket /run/podman/podman.sock http://foo/_ping
OK

$ limactl shell podman-rootful ls -l /run/podman/podman.sock
ls: cannot access '/run/podman/podman.sock': Permission denied

$ limactl shell podman-rootful sudo ls -l /run/podman/podman.sock
srw-rw----. 1 root root 0 Dec 25 16:13 /run/podman/podman.sock
```

Fedora 40:
```console
$ limactl shell podman-rootful-f40 curl --unix-socket /run/podman/podman.sock http://foo/_ping
OK

$ limactl shell podman-rootful-f40 ls -l /run/podman/podman.sock
srw-rw----. 1 suda suda 0 Dec 25 16:18 /run/podman/podman.sock
```

Something seems wrong in:
https://github.com/lima-vm/lima/blob/5fb93532e6cf9edd7201815475d9d13d97561444/templates/podman-rootful.yaml#L32-L46

**Comment by AkihiroSuda** [comment]
`/var/log/cloud-init-output.log` differs.

Fedora 41:
```
LIMA 2024-12-25T16:13:34+09:00| Executing /mnt/lima-cidata/provision.system/00000000
+ command -v podman
+ exit 0
LIMA 2024-12-25T16:13:34+09:00| Executing /mnt/lima-cidata/provision.system/00000001
+ systemctl --system enable --now podman.socket
Created symlink '/etc/systemd/system/sockets.target.wants/podman.socket' → '/usr/lib/systemd/system/podman.socket'.
LIMA 2024-12-25T16:13:34+09:00| Exiting with code 0
```

Fedora 40:
```
LIMA 2024-12-25T16:18:12+09:00| Executing /mnt/lima-cidata/provision.system/00000000                                                                             
+ command -v podman                                                                                                                                              
+ '[' '!' -e /etc/systemd/system/podman.socket.d/override.conf ']'                                                                                               
+ mkdir -p /etc/systemd/system/podman.socket.d                                                                                                                   
+ cat                                                                                                                                                            
+ '[' '!' -e /etc/tmpfiles.d/podman.conf ']'                                                                                                                     
+ mkdir -p /etc/tmpfiles.d                                                                                                                                       
+ echo 'd /run/podman 0700 suda -'                                                                                                                               
+ dnf -y install podman
[...]
Complete!
LIMA 2024-12-25T16:18:52+09:00| Executing /mnt/lima-cidata/provision.system/00000001
+ systemctl --system enable --now podman.socket
Created symlink /etc/systemd/system/sockets.target.wants/podman.socket → /usr/lib/systemd/system/podman.socket.
LIMA 2024-12-25T16:18:52+09:00| Exiting with code 0
```

**Comment by afbjorklund** [comment]
This should have been fixed with the previous commit ab544b5c12bc359beda7866f7101759b018f123d
